src/crawlee/_utils/models.py

```diff
@@ -9,20 +9,17 @@
 """Utility types for Pydantic models."""
 
 
-def _timedelta_to_ms(td: timedelta | None) -> Any:
+def _timedelta_to_ms(td: timedelta) -> float:
     if td == timedelta.max:
         return float('inf')
 
-    if td is None:
-        return td
-
     return int(round(td.total_seconds() * 1000))
 
 
 _number_parser = TypeAdapter(float)
 
 
-def _timedelta_from_ms(value: float | timedelta | Any | None, handler: Callable[[Any], Any]) -> Any:
+def _timedelta_from_ms(value: float | timedelta | Any | None, handler: Callable[[Any], timedelta]) -> timedelta | None:
```

Comment:
Was this the problematic type hint? :slightly_smiling_face:

Reply:
I think it is still Pydantic bug, somewhere down there. I did not spent effort on trying to figure it out once I saw it is working well with more specific model.
I can reproduce it if I use new Pydantic version and for example in the new tests I created I do not import **Any** in the test file. It will then fail on importing **Any** even if I do not use it in the test files and it is needed only in model file.

Reply:
It basically goes all the way to problem with the way annotations are evaluated: https://github.com/pydantic/pydantic/issues/2678

When you use **from __future__ import annotations** , it can break pydantic :D in those files

Reply:
That sounds scary, future annotations should be the norm 😃 can you elaborate a bit about what the problem is?

Reply:
I do not think I can elaborate. This requires knowledge I do not posses. Let's wait with this then and see if Pydantic people can explain it: https://github.com/pydantic/pydantic/issues/11004